py/selenium/webdriver/webkitgtk/service.py

```diff
@@ -53,18 +53,30 @@ def __init__(
         **kwargs,
     ) -> None:
         self._service_args = list(service_args or [])
-        if log_path is not None:
-            warnings.warn("log_path is deprecated, use log_output instead", DeprecationWarning, stacklevel=2)
-            log_path = open(log_path, "wb")
-        log_output = open(log_output, "wb") if log_output else None
 
-        super().__init__(
-            executable_path=executable_path,
-            port=port,
-            log_output=log_path or log_output,
-            env=env,
-            **kwargs,
-        )
+        log_file: Optional[IO[bytes]] = None
+
+        try:
+            if log_output is not None:
+                log_file = open(log_output, "wb")
+            elif log_path is not None:
+                warnings.warn("log_path is deprecated, use log_output instead", DeprecationWarning, stacklevel=2)
+                log_file = open(log_path, "wb")
+
+            super().__init__(
+                executable_path=executable_path,
+                port=port,
+                log_output=log_file,
+                env=env,
+                **kwargs,
+            )
+        except Exception:
+            if log_file is not None:
+                try:
+                    log_file.close()
+                except Exception:
+                    pass
+            raise
```

Comment:
wrapping everything in one big try/except like this just to possibly close a file handle (that will get closed anyway) is really ugly and unnecessary.